cmd/kube-apiserver/app/server.go

```diff
@@ -511,7 +516,24 @@ func getRuntimeConfigValue(s *options.APIServer, apiKey string, defaultValue boo
 }
 
 // Parses the given runtime-config and formats it into map[string]ApiGroupVersionOverride
-func parseRuntimeConfig(s *options.APIServer) (map[string]genericapiserver.APIGroupVersionOverride, error) {
+func parseRuntimeConfig(s *options.APIServer) (genericapiserver.APIResourceEnablement, error) {
+	v1GroupVersionString := "api/v1"
+	v1GroupVersion := unversioned.GroupVersion{Group: api.GroupName, Version: "v1"}
+	extensionsGroupVersionString := "extensions/v1beta1"
+	extensionsGroupVersion := unversioned.GroupVersion{Group: extensions.GroupName, Version: "v1beta1"}
+	batchGroupVersionString := "batch/v1"
+	batchGroupVersion := unversioned.GroupVersion{Group: batch.GroupName, Version: "v1"}
+	autoscalingGroupVersionString := "autoscaling/v1"
+	autoscalingGroupVersion := unversioned.GroupVersion{Group: autoscaling.GroupName, Version: "v1"}
+	versionToEnablementString := map[unversioned.GroupVersion]string{
+		v1GroupVersion:          v1GroupVersionString,
+		extensionsGroupVersion:  extensionsGroupVersionString,
+		batchGroupVersion:       batchGroupVersionString,
+		autoscalingGroupVersion: autoscalingGroupVersionString,
+	}
+
+	resourceEnablementConfig := master.DefaultResourceEnablement()
+
 	// "api/all=false" allows users to selectively enable specific api versions.
 	disableAllAPIs := false
```

Comment:
> might read easier below as:

I'm not sure that anything can help logic where specifying `api/all=false` can result in **some** "special" APIs being enabled, but only if you really want them and sometimes having flags that are `group/version`, sometimes `magicstring`, sometimes `magicstring/version`, and sometimes `magicstring/othermagicstring`.  Along with overall enablement being sometimes respected per resources, sometimes on by default, sometimes off by default, and sometimes respected per version only.

And don't forget that `magicstring/othermagicstring` doesn't respect the default set by `magicstring`, but all the other ones do.